1Homework18.02.22/SquareSorting/SquareSorting/Program.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+
+namespace SquareSorting;
+internal class Program
+{
+    /// <summary>
+    /// Checks if an array is sorted
+    /// </summary>
+    /// <param name="array">the sortable array</param>
+    /// <returns>true if the array is sorted</returns>
+    private static bool IsArraySorted(int[] array)
+    {
+        for (int i = 0; i < array.Length - 1; ++i)
+        {
+            if (array[i] > array[i + 1])
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+    
+    /// <summary>
+    /// Swaps two integer elements
+    /// </summary>
+    /// <param name="firstVariable"></param>
+    /// <param name="secondVariable"></param>
+    private static void Swap(ref int firstVariable, ref int secondVariable)
+    {
+        firstVariable ^= secondVariable;
+        secondVariable ^= firstVariable;
+        firstVariable ^= secondVariable;
+    }
+    
+    /// <summary>
+    /// Sorts an array by Bubble sorting
+    /// </summary>
+    /// <param name="array">the sortable array</param>
+    private static void SortByBubble(int[] array)
+    {
+        for (int i = 0; i < array.Length; ++i)
+        {
+            for (int j = 0; j < array.Length - 1 - i; ++j)
+            {
+                if (array[j] > array[j + 1])
+                {
+                    Swap(ref array[j], ref array[j + 1]);
+                }
+            }
+        }
+    }
+
+    private static bool TestAreArraySorted()
+    {
+        var arrayTest1 = new[] { 5, 7, 4, -8, 1, 5 };
+        var arrayTest2 = new [] { 0 };
+        var arrayTest3 = new[] { -1, 2, 3, 4, 5 };
+
+        return !IsArraySorted(arrayTest1) && IsArraySorted(arrayTest2)
+                                          && IsArraySorted(arrayTest3);
```

Comment:
Избегайте выравнивания, оно ломается при рефакторинге
```suggestion
        return !IsArraySorted(arrayTest1) && IsArraySorted(arrayTest2)
            && IsArraySorted(arrayTest3);
```